server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -160,30 +159,13 @@ public static Query termsQuery(String column, DataType<?> type, Collection<?> va
             return null;
         }
         StorageSupport<?> storageSupport = type.storageSupport();
-        EqQuery<?> eqQuery = storageSupport == null ? null : storageSupport.eqQuery();
-        if (eqQuery == null) {
-            return booleanShould(column, type, nonNullValues, hasDocValues, indexType);
```

Comment:
In the end, booleanShould also returns null(fall back to generic query) if eqQuery == null.